### PR TITLE
Don't serialize shorthand when not all subproperties have initial values

### PR DESCRIPTION
**Pull request by cdoublev**
This test case corresponds to a current round trip issue in Chrome/Firefox (at least): grid serializes to 'none' even if grid-template-areas is declared to a non-initial value.

### CODE REVIEW

File: css/css-grid/parsing/grid-shorthand-serialization.html [hunk 1]
```diff
@@ -20,6 +20,14 @@
     assert_equals(element.style.gridTemplateAreas, "none");
   }, "Serialization without grid-template-areas");
 
+  test(() => {
+    const gridTemplateAreasValue = '"one"';
+    element.style.grid = 'none'
+    element.style.gridTemplateAreas = gridTemplateAreasValue;
+    assert_equals(element.style.grid, "", "grid shorthand must not be serialized when grid-template-areas is set to a non-initial value");
```
Comment (Loirooriol) [inline css/css-grid/parsing/grid-shorthand-serialization.html:27]: Can you check `element.style.gridTemplate` too?

**Review by Loirooriol — APPROVED** [review]
While you are at it, do you want to test the more complex case from https://bugs.chromium.org/p/chromium/issues/detail?id=1322084 ? And https://bugs.chromium.org/p/chromium/issues/detail?id=1353822#c7 has cases for similar problems. But no need to do it in this PR.

**Comment by Loirooriol** [comment]
> This test case corresponds to a current round trip issue in Chrome/Firefox

Firefox and WebKit seem fine for me. Just Blink is wrong

**Comment by cdoublev** [comment]
Sure. I have already spent some time analyzing the conditions for which `grid` or `grid-template` should not be serialized, and to figure out how to formulate these conditions in the most simple terms. However it does not surprise me that I missed `grid: auto-flow 1px/2px; grid-auto-columns: 3px` as I had decided to come back later to it with a fresh look. Furthermore, I was quite confident for this simple test but ideally I should run new tests locally and I might struggle to do so as I am using a virtual machine to keep my host machine "clean". But I can still open the HTML test file directl and running `wpt lint` should be enough, of course.

TLDR: I will do it later but feel free to replace this PR if you do not want to wait.

File: css/css-grid/parsing/grid-shorthand-serialization.html (hunk 1 above)
Comment (cdoublev) [reply css/css-grid/parsing/grid-shorthand-serialization.html:27]: In this test or in a similar test in `/css/css-grid/parsing/grid-template-shorthand.html`?

Comment (Loirooriol) [reply css/css-grid/parsing/grid-shorthand-serialization.html:27]: Strictly speaking grid-template-shorthand.html may be more suitable, but a oneliner here may be simpler and some tests below are already checking `gridTemplate`. Up to you

**Comment by cdoublev** [comment]
Superseded by #36217

Oriol (or anyone interested), I leave you my notes in case you want to review/complete tests about serializing `grid`/`grid-template` to empty string when it cannot represent all its sub-properties.

> **`grid-template`**
>
> It is assumed that `grid-template-rows`, `grid-template-columns`, `grid-template-areas`, have a declaration.
>
> `grid-template` cannot represent all its sub-properties when `grid-template-areas` is not declared to its initial value (`none`) and some of these conditions is truthy:
>
>   - `grid-template-rows` matches `none`
>   - `grid-template-rows` defines less/more `<track-size>`s than `<string>` (rows) in `grid-template-areas`
>   - `grid-template-rows|columns` matches `<track-list>` expanded with `<track-repeat>`
>   - `grid-template-rows|columns` matches `<auto-track-list>`
>   - `grid-template-rows|columns` matches `subgrid <line-name-list>?`
>
> Alternatively, the last three conditions can be expressed as follows to guide implementation rather than tests:
>
>   - `grid-template-rows` matches `<track-list>` expanded with `<track-size>`
>   - `grid-template-columns` matches either `none`, `<track-list>` expanded with `<track-size>` (explicit track list), or `<explicit-track-list>`
>
> **`grid`**
>
> It is assumed that `grid-template-rows`, `grid-template-columns`, `grid-template-areas`, `grid-auto-fow`, `grid-auto-rows`, `grid-auto-columns`, have a declaration.
>
> `grid` cannot represent all its sub-properties when some of these conditions is truthy:
>
>   - some `grid-template-*` sub-property is not declared to its initial value and the serialization of `grid-template` is empty string
>   - both axis have declarations defining an implicit track list, which means some of the following conditions is truthy:
>     - `grid-auto-column` is not declared to its initial value (`auto`)
>     - `grid-auto-flow` is declared to a value including `column`
>   - some axis has declarations defining both an explicit and implicit track lists, which means some of the following conditions is truthy:
>     - `grid-auto-flow` is not declared to its initial value (`row`) and some `grid-template-*` sub-property is not declared to its initial value
>     - `grid-template-areas` is not declared to its initial value (`none`) and some `grid-auto-*` sub-property is not declared to its initial value
>     - `grid-template-rows` and `grid-auto-rows` are both not declared to their initial value (`none` and `auto`)
>     - `grid-template-columns` and `grid-auto-columns` are both not declared to their initial value (`none` and `auto`)
